src/hook/useStepper.spec.ts

```diff
@@ -0,0 +1,167 @@
+    .set('secondStep', 'uncompleted')
+    .set('thirdStep', 'disabled')
+    .set('fourthStep', 'uncompleted')
+
+  const state2_previousClicked = {
+    currentStep: 'firstStep',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('firstStep', 'completed')
+      .set('secondStep', 'uncompleted')
+      .set('thirdStep', 'disabled')
+      .set('fourthStep', 'uncompleted'),
+    enabledSteps: ImmutableList(['firstStep', 'secondStep', 'fourthStep'])
+  }
+
+  const state2_stepCompleted = {
+    currentStep: 'fourthStep',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('firstStep', 'completed')
+      .set('secondStep', 'completed')
+      .set('thirdStep', 'disabled')
+      .set('fourthStep', 'uncompleted'),
+    enabledSteps: ImmutableList(['firstStep', 'secondStep', 'fourthStep'])
+  }
+
+  const stepStatus3 = OrderedMap<StepId, StepStatus>()
+    .set('firstStep', 'disabled')
+    .set('secondStep', 'uncompleted')
+    .set('thirdStep', 'uncompleted')
+    .set('fourthStep', 'uncompleted')
+    .set('fifthStep', 'disabled')
+
+  const state3_previousClicked = {
+    currentStep: 'secondStep',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('firstStep', 'disabled')
+      .set('secondStep', 'uncompleted')
+      .set('thirdStep', 'uncompleted')
+      .set('fourthStep', 'uncompleted')
+      .set('fifthStep', 'disabled'),
+    enabledSteps: ImmutableList(['secondStep', 'thirdStep', 'fourthStep'])
+  }
+
+  const state3_stepCompleted = {
+    currentStep: 'fourthStep',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('firstStep', 'disabled')
+      .set('secondStep', 'uncompleted')
+      .set('thirdStep', 'completed')
+      .set('fourthStep', 'uncompleted')
+      .set('fifthStep', 'disabled'),
+    enabledSteps: ImmutableList(['secondStep', 'thirdStep', 'fourthStep'])
+  }
+
+  const stepStatus4 = OrderedMap<StepId, StepStatus>()
+    .set('step1', 'completed')
+    .set('step2', 'completed')
+    .set('step3', 'uncompleted')
+
+  const state4_stepperSubmitted: StepperState = {
+    currentStep: 'step3',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('step1', 'completed')
+      .set('step2', 'completed')
+      .set('step3', 'completed'),
+    enabledSteps: ImmutableList(['step1', 'step2', 'step3'])
+  }
+
+  const state4_initializerArgs = {
+    initialActiveStep: 'step1',
+    initialStatus: OrderedMap<StepId, StepStatus>()
+      .set('step1', 'uncompleted')
+      .set('step2', 'uncompleted')
+      .set('step3', 'uncompleted')
+  }
+
+  const state4_stepperReseted: StepperState = {
+    currentStep: 'step1',
+    stepStatus: OrderedMap<StepId, StepStatus>()
+      .set('step1', 'uncompleted')
+      .set('step2', 'uncompleted')
+      .set('step3', 'uncompleted'),
+    enabledSteps: ImmutableList(['step1', 'step2', 'step3'])
+  }
+
+  describe.each`
+    current         | status          | type                | payload                   | expectedResult
```

Comment:
What does `current` means? 🤔

The data parameters must be the most explicit possible, if not the test will be difficult to understand.

I will have used an object `Data` to type parameters with explicit naming: 

```ts
type Data = {
  stepId: string
  stepsStatuses: OrderedMap<StepId, StepStatus>
  actionType: string
  actionPayload: InitializerArgs
  expectedState: StepperState
}
```